src/main/drivers/accgyro_mpu.c

```diff
@@ -234,8 +234,7 @@ static bool detectSPISensorsAndUpdateDetectionResult(gyroDev_t *gyro, const sens
 #else
     UNUSED(bus);
 #endif
-    if (mpu6000SpiDetect(&gyro->bus)) {
-        gyro->mpuDetectionResult.sensor = MPU_60x0_SPI;
+    if ((gyro->mpuDetectionResult.sensor = mpu6000SpiDetect(&gyro->bus))) {
```

Comment:
For MPU6500 variants, `mpuDetectionResult.sensor` will always be MPU_65xx_SPI at this point. `gyroHardware` will be `GYRO_MPU6500`, and corresponding element in `detectedSensors` will be `GYRO_MPU6500`.

Which will in turn result in `status` command to display MPU6500 for ICM_20602_SPI and ICM_20608_SPI cases. There seems to be no other places using the value in `mpuDetectionResult.sensor`, so the loss of resolution of the device is only the impact of not honoring the result of `mpu6500SpiDetect`. I personally would like to preserve it, though.